protocol/src/main/kotlin/tbdex/sdk/protocol/models/Rfq.kt

```diff
@@ -68,31 +95,117 @@ class Rfq private constructor(
     // TODO check that VCs satisfying PD are crypto verified
 
     try {
-      PresentationExchange.satisfiesPresentationDefinition(data.claims, requiredClaims)
+      PresentationExchange.satisfiesPresentationDefinition(
+        privateData?.claims ?: emptyList(),
+        requiredClaims
+      )
     } catch (e: Exception) {
       throw IllegalArgumentException("No matching claim for Offering requirements: ${requiredClaims.id}")
     }
   }
 
+  /**
+   * Verify the presence and integrity of all possible properties in Rfq.privateData
+   * @throws Exception if there are properties missing in Rfq.privateData or which do not match the corresponding
+   *                   hashed property in Rfq.data
+   */
+  fun verifyAllPrivateData() {
+    // Verify payin details
+    data.payin.paymentDetailsHash?.let {
+      verifyPayinDetailsHash()
+    }
+
+    // Verify payout details
+    data.payout.paymentDetailsHash?.let {
+      verifyPayoutDetailsHash()
+    }
+
+    // Verify claims
+    data.claimsHash?.let {
+      verifyClaimsHash()
+    }
+  }
+
+  /**
+   * Verify the integrity properties that are present in
+   * @throws Exception if there are properties present in Rfq.privateData which do not match the corresponding
+   *                   hashed property in Rfq.data
+   */
+  fun verifyPresentPrivateData() {
+    privateData ?: throw Error("privateData property is missing")
```

Comment:
better to throw IllegalArgument or IllegalStateException i think

Reply:
After looking closer, this check should actually happen within each of `verifyPayinDetailsHash()`, `verifyPayoutDetailsHash()`, and `verifyClaimsHash()`, because it's possible that the particular `Rfq` does not have any private data